src/Compute/Compute/Generated/Snapshot/SnapshotCreateOrUpdateMethod.cs

```diff
@@ -41,6 +41,21 @@ public override void ExecuteCmdlet()
             base.ExecuteCmdlet();
             ExecuteClientAction(() =>
             {
+                Snapshot existingResource;
+                try
+                { 
+                    existingResource = SnapshotsClient.Get(this.ResourceGroupName, this.SnapshotName);
+                }
+                catch
+                {
+                    existingResource = null;
+                }
+
+                if (existingResource != null)
+                {
+                    throw new Exception(string.Format("A Snapshot with name '{0}' in resource group '{1}' already exists. Please use Update-AzSnapshot to update an existing Snapshot.", this.SnapshotName, this.ResourceGroupName));
+                }
+
                 if (ShouldProcess(this.SnapshotName, VerbsCommon.New))
```

Comment:
looks good. this should do the job. But I wonder how other resources check for existing resource

Reply:
I don't think this pattern is commonly used yet, despite it being a [BP](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/new-cmdlet-example.md#new--cmdlet-example).